tests/module_api/test_api.py

```diff
@@ -532,6 +534,38 @@ def test_update_membership(self):
         self.assertEqual(res["displayname"], "simone")
         self.assertIsNone(res["avatar_url"])
 
+        # Check that no remote join attempts have occurred thus far.
+        self.assertFalse(mocked_remote_join.called)
+
+        # Necessary to fake a remote join.
+        fake_stream_id = 1
+        if isinstance(mocked_remote_join, MagicMock):
+            # AyncMock is not supported in this Python version.
```

Comment:
What does this mean? I don't see anything in https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock that says any of the versions of Python we support in Synapse doesn't support `MagicMock`s.